chainer/datasets/tox21.py

```diff
@@ -0,0 +1,150 @@
+import os
+import shutil
+import warnings
+import zipfile
+
+
+from chainer.dataset import download
+from chainer.datasets import tuple_dataset
+import numpy
+
+try:
+    from rdkit import Chem
+    from rdkit.Chem import rdMolDescriptors
+    _available = True
+except ImportError:
+    _available = False
+
+
+def check_available():
+    """Checks availability of Tox21 dataset.
+
+    This function checks the availability of Tox21 dataset
+    in user's environment.
+    Specifically, we use `RDKit <https://github.com/rdkit/rdkit>`_
+    to extract features and labels from raw files, whose format are
+    `SDF <https://en.wikipedia.org/wiki/Chemical_table_file#SDF>`_.
+    So, it returns ``True`` if Chainer successfully imports the
+    RDKit module.
+
+    Returns:
+        ``True`` is Tox21 dataset is available.
+        Otherwise ``False``.
+
+    """
+    if not _available:
+        warnings.warn('rdkit is not install on your environment '
+                      'Please install it to use tox21 dataset.\n'
+                      'See the official document for installation.'
+                      'http://www.rdkit.org/docs/Install.html')
+    return _available
+
+
+config = {
+    'train': {
+        'url': 'https://tripod.nih.gov/tox21/challenge/download?'
+        'id=tox21_10k_data_allsdf',
+        'filename': 'tox21_10k_data_all.sdf'
+    },
+    'val': {
+        'url': 'https://tripod.nih.gov/tox21/challenge/download?'
+        'id=tox21_10k_challenge_testsdf',
+        'filename': 'tox21_10k_challenge_test.sdf'
+    },
+    'test': {
+        'url': 'https://tripod.nih.gov/tox21/challenge/download?'
+        'id=tox21_10k_challenge_scoresdf',
+        'filename': 'tox21_10k_challenge_score.sdf'
+    }
+}
+
+root = 'pfnet/chainer/tox21'
+
+label_names = ['NR-AR', 'NR-AR-LBD', 'NR-AhR', 'NR-Aromatase', 'NR-ER',
```

Comment:
Global variables `config`, `root` and `label_names` are for internal use. You shoud prefix them with `_`.